url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwInvest/output/Performance.java

```diff
@@ -0,0 +1,254 @@
+	private long start;
+	private long end;
+	private Plot cash, balance, sentiment, buildup, zero, threshold;
+	private Plot inverseThreshold, position, buy, sell;
+	private Individual individual;
+	private Settings settings;
+	
+	/**
+	 * Instantiate a new bar-chart
+	 * @param individual the individual being tested
+	 * @param settings	 the settings being applied
+	 * @param ticker	 the ticker object
+	 * @param start		 the start time
+	 * @param end		 the end time
+	 */
+	public Performance(Individual individual, Settings settings, long start, long end) {		
+		setHeightProportion(1);
+		setWidthProportion(1);
+		setGridColor(new Color(200,200,200));
+		setVerticalLines(21);
+		setHorizontalLines(11);
+		
+		this.individual = individual;
+		this.settings = settings;
+		
+		cash = new Plot(new ArrayList<Point2D.Double>());
+		cash.setSize(0);
+		cash.setConnect(true);
+		cash.setConnectColor(new Color(0,192,0));
+		addPlot(cash);
+		
+		balance = new Plot(new ArrayList<Point2D.Double>());
+		balance.setSize(0);
+		balance.setConnect(true);
+		balance.setConnectColor(new Color(0,0,255));
+		addPlot(balance);
+		
+		buildup = new Plot(new ArrayList<Point2D.Double>());
+		buildup.setSize(0);
+		buildup.setConnect(true);
+		buildup.setConnectColor(new Color(255,0,255));
+		addPlot(buildup);
+		
+		sentiment = new Plot(new ArrayList<Point2D.Double>());
+		sentiment.setSize(4);
+		sentiment.setFill(false);
+		sentiment.setConnect(false);
+		sentiment.setCircle(true);
+		sentiment.setPointColor(new Color(0,0,0));
+		addPlot(sentiment);
+		
+		zero = new Plot(new ArrayList<Point2D.Double>());
+		zero.setSize(0);
+		zero.setConnect(true);
+		zero.setConnectColor(new Color(100,100,100));
+		addPlot(zero);
+		
+		threshold = new Plot(new ArrayList<Point2D.Double>());
+		threshold.setSize(0);
+		threshold.setConnect(true);
+		threshold.setConnectColor(new Color(0,255,0));
+		addPlot(threshold);
+		
+		inverseThreshold = new Plot(new ArrayList<Point2D.Double>());
+		inverseThreshold.setSize(0);
+		inverseThreshold.setConnect(true);
+		inverseThreshold.setConnectColor(new Color(255,0,0));
+		addPlot(inverseThreshold);
+	
+		position = new Plot(new ArrayList<Point2D.Double>());
+		position.setSize(0);
+		position.setConnect(true);
+		position.setConnectColor(new Color(0,180,0));
+		addPlot(position);
+		
+		buy = new Plot(new ArrayList<Point2D.Double>());
+		buy.setSize(5);
+		buy.setFill(true);
+		buy.setConnect(false);
+		buy.setCircle(false);
+		buy.setPointColor(new Color(0,255,0));
+		addPlot(buy);
+		
+		sell = new Plot(new ArrayList<Point2D.Double>());
+		sell.setSize(5);
+		sell.setFill(true);
+		sell.setConnect(false);
+		sell.setCircle(false);
+		sell.setPointColor(new Color(255,0,0));
+		addPlot(sell);
+		
+		setRange(start, end);
+	}
+	
+	/**
+	 * Set the time range, and recalculate the range
+	 * @param start the starting time 
+	 * @param end   the ending time
+	 */
+	public void setRange(long start, long end) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 48 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 144-158, lines 161-202, lines 205-227 into separate methods.